lib/koans/15_protocols.ex

```diff
@@ -0,0 +1,39 @@
+defmodule Protocols do
+  use Koans
+
+  defprotocol School do
+    def enrol(name)
+  end
+
+  defimpl School, for: SecondarySchool do
+    def enrol(name) do
+      "#{name} enrolled at secondary school"
+    end
+  end
+
+  defimpl School, for: DanceSchool do
+    def enrol(name) do
+      "#{name} enrolled for dance"
+    end
+  end
+
+  defmodule EveningClass do end
+
+  @schools %{
+    :secondary => School.SecondarySchool,
```

Comment:
Again you are right, however I was trying to make the user think a little bit. There are two 'Schools' that implement the Protocol, and I was hoping the user might have to figure out the answer. Directly calling the module as you suggested earlier or just having one entry in the map, makes the answer quite obvious and doesn't really show the polymorphic advantage. 

I could add another test similar to the one for DanceSchool, for SecondarySchool? Then both would be used..